src/hera.cpp

```diff
@@ -50,6 +52,49 @@ struct hera_instance : evm_instance {
 
 namespace {
 
+#if HERA_EVM2WASM_JS
+// NOTE: assumes that pattern doesn't contain any formatting characters (e.g. %)
+string mktemp_string(string pattern) {
+  char tmp[PATH_MAX] = { 0 };
+  size_t len = snprintf(tmp, sizeof(tmp), "%s", pattern.c_str());
+  mktemp(tmp);
```

Comment:
This will not work.
1. The mktemp expects exactly 6 XXXXXX.
2. mktemp() docs suggest to use mkstemp() http://man7.org/linux/man-pages/man3/mkstemp.3.html.
3. You have to check the returned value.

Reply:
I am not sure, my man page says:
> the number of unique file names mktemp() can return depends on the number of `Xs' provided;

Reply:
Seems like on BSD it is flexible, but Linux is more picky:
> The last six characters of template must be "XXXXXX" and these are replaced with a string that makes the filename unique.

Reply:
This should be fine then:
```
string makeTempPath(string pattern) {
  mktemp(pattern.data());
  return pattern;
}
```

Reply:
True, just make `pattern` pass by value.